service/frontend/errors.go

```diff
@@ -72,7 +72,6 @@ var (
 	errStatusFilterMustBeNotRunning                       = serviceerror.NewInvalidArgument("StatusFilter must be specified and must be not Running.")
 	errCronNotAllowed                                     = serviceerror.NewInvalidArgument("Scheduled workflow must not contain CronSchedule")
 	errIDReusePolicyNotAllowed                            = serviceerror.NewInvalidArgument("Scheduled workflow must not contain WorkflowIDReusePolicy")
-	errShuttingDown                                       = serviceerror.NewUnavailable("Shutting down")
```

Comment:
> since the window it would apply is basically nil.

I definitely think we should get rid of this error. However, I want to ensure that it's guaranteed it will never occur. Are we adding start and stop lifecycle hooks to `WorkflowHandler` and ensuring that it's only constructed through fx? That would certainly help. 

However, even that isn't a guarantee because dependents of this object may not respect the lifecycle and zombie goroutines may continue to use the handler. 

I think ensuring we dependents may only obtain a workflow handler after lifecycle hooks are registered is an acceptable compromise, though, because using any object outside of its lifecycle is always misuse, and we shouldn't need to support dependents that do this. As long as we use fx, don't start any goroutines before start, and close all goroutines before returning from the stop hook, this won't happen.

If you want to be extra safe for this change, I think we should restructure our code to ensure that goroutines exist only within the lifecycle of the object that spawned them, which is something we should do any way. However, that's a lot of work, so definitely optional for this change.

Reply:
> I definitely think we should get rid of this error. However, I want to ensure that it's guaranteed it will never occur. Are we adding start and stop lifecycle hooks to `WorkflowHandler` and ensuring that it's only constructed through fx? That would certainly help.

I don't see anything that holds a reference to the WorkflowHandler besides the Service (which is managing the shutdown) and grpc itself (which we're asking to shut down). Do you?

But also, WorkflowHandler.Stop doesn't actually _do_ anything (that's not already been done), so it doesn't really matter if Stop has been called or not. There's no point in paying attention to it.

Reply:
In that case, it seems fine to me.